drivers/flash/flash_mcux_flexspi.c

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2020 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT	nxp_imx_flexspi
+
+#include <drivers/flash.h>
+#include <logging/log.h>
+#include <sys/util.h>
+#include "flash_mcux_flexspi.h"
+
+LOG_MODULE_REGISTER(flash_flexspi, CONFIG_FLASH_LOG_LEVEL);
+
+struct flash_flexspi_config {
+	FLEXSPI_Type *base;
+	uint8_t *ahb_base;
+	bool ahb_bufferable;
+	bool ahb_cacheable;
+	bool ahb_prefetch;
+	bool ahb_read_addr_opt;
+	bool combination_mode;
+	bool xip;
+	flexspi_read_sample_clock_t rx_sample_clock;
```

Comment:
As an enum this could be `sizeof(int)`; some alignment costs might be avoided by moving it above the `bool` fields.